app/components/at_mention/at_mention.js

```diff
@@ -159,7 +159,7 @@ export default class AtMention extends React.PureComponent {
                 mention = group.name;
                 suffix = this.props.mentionName.substring(group.name.length);
             } else {
-                const pattern = new RegExp(/(all|channel|here)\.?/, 'i');
+                const pattern = new RegExp(/\b(all|channel|here)(\.|_)?\b/, 'i');
```

Comment:
The `\b` at the end is incorrect. If it's after a period, it would be `\B` since there wouldn't be a boundary between a period and whiteespace. We'd still want a `\b` after the underscore though since it's considered a word character.

Reply:
Maybe this would work?
```suggestion
                const pattern = new RegExp(/\b(all|channel|here)(?:\.\B|_\b|\b)/, 'i');
```
Either it's followed by a period and more non-word characters, an underscore and non-word characters, or any non-word character.

Reply:
Thanks for this @hmhealey